capi/include/hyper.h

```diff
@@ -83,41 +127,86 @@ typedef enum {
   HYPER_TASK_BUF,
 } hyper_task_return_type;
 
-typedef struct hyper_executor hyper_executor;
-
-typedef struct hyper_io hyper_io;
-
-typedef struct hyper_task hyper_task;
-
+/*
+ A streaming HTTP body.
+ */
 typedef struct hyper_body hyper_body;
 
+/*
+ A buffer of bytes that is sent or received on a `hyper_body`.
+ */
 typedef struct hyper_buf hyper_buf;
 
+/*
+ An HTTP client connection handle.
+
+ These are used to send a request on a single connection. It's possible to
+ send multiple requests on a single connection, such as when HTTP/1
+ keep-alive or HTTP/2 is used.
+ */
 typedef struct hyper_clientconn hyper_clientconn;
 
+/*
+ An options builder to configure an HTTP client connection.
+ */
 typedef struct hyper_clientconn_options hyper_clientconn_options;
 
+/*
+ An async context for a task that contains the related waker.
+ */
 typedef struct hyper_context hyper_context;
 
+/*
+ A more detailed error object returned by some hyper functions.
+ */
 typedef struct hyper_error hyper_error;
 
+/*
+ A task executor for `hyper_task`s.
+ */
+typedef struct hyper_executor hyper_executor;
+
+/*
+ An HTTP header map.
+
+ These can be part of a request or response.
+ */
 typedef struct hyper_headers hyper_headers;
 
+/*
+ An IO object used to represent a socket or similar concept.
+ */
+typedef struct hyper_io hyper_io;
+
+/*
+ An HTTP request.
+ */
 typedef struct hyper_request hyper_request;
 
+/*
+ An HTTP response.
+ */
 typedef struct hyper_response hyper_response;
 
+/*
+ An async task.
+ */
+typedef struct hyper_task hyper_task;
+
+/*
+ A waker that is saved and used to waken a pending task.
+ */
 typedef struct hyper_waker hyper_waker;
 
-typedef int (*hyper_body_foreach_callback)(void*, const hyper_buf*);
+typedef int (*hyper_body_foreach_callback)(void*, const struct hyper_buf*);
```

Comment:
Hm, is this change from an updated version of cbindgen? What's the repercussions of this change? Will users (like curl) need to change their type annotations? Or is it just the same thing before and after?

Reply:
Oh gosh, I have no idea. Let me ask some folks who understand C better than me

Reply:
I generated that using the latest `cbindgen` and same `cbindgen.toml` as previous. I can look through the cbindgen docs to see if they mention changes.

For what it's worth, the curl maintainer said he's OK treating Hyper support as bleeding edge, which implies he's not super concerned about backwards compatibility.

Reply:
Cool, thanks for looking! I'm not annoyed by the change or anything, I just want to better understand before we merge :)

Reply:
My coworker - who has written a lot of C - says they're equivalent. I also tested compiling Curl with this new hyper.h and it generated with no issues. See https://github.com/meterup/homebrew-safe for details or `brew install meterup/safe/hyper meterup/safe/curl`.

Reply:
Cool, thanks for checking :D